src/test/java/com/timgroup/statsd/UnixSocketTest.java

```diff
@@ -159,16 +192,26 @@ public void resist_dsd_timeout() throws Exception {
 
         // Freeze the server to simulate dsd being overwhelmed
         server.freeze();
-        while(lastException.getMessage() == null) {
-            client.gauge("mycount", 20);
-            Thread.sleep(10);  // We need to fill the buffer, setting a shorter sleep
+        if (transport == "unixgram") {
+            while (lastException.getMessage() == null) {
+                client.gauge("mycount", 20);
+                Thread.sleep(10);  // We need to fill the buffer, setting a shorter sleep
+            }
+            String excMessage = TestHelpers.isLinux() ? "Resource temporarily unavailable" : "No buffer space available";
+            assertThat(lastException.getMessage(), containsString(excMessage));
+        } else {
+            // We can't realistically fill the buffer on a stream socket because we can't timeout in the middle of packets,
+            // so we just send a lot of packets
```

Comment:
Does this actually test anything? Would it make sense to run this test only for datagram sockets?

Reply:
This actually can work with the new write logic